packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx

```diff
@@ -0,0 +1,165 @@
+import { TextField, Button, Typography } from '@mui/material'
+import { useState, useMemo } from 'react'
+import { EthereumTokenType, useFungibleTokenBalance, useWeb3, useAccount } from '@masknet/web3-shared-evm'
+import { useI18N } from '../../../utils'
+import { useStyles } from './SavingsFormStyles'
+import { IconURLS } from './IconURL'
+import { TabType } from '../types'
+import { SavingsProtocols } from '../protocols'
+import { isLessThan, leftShift, rightShift } from '@masknet/web3-shared-base'
+import { EthereumWalletConnectedBoundary } from '../../../web3/UI/EthereumWalletConnectedBoundary'
+import { EthereumChainBoundary } from '../../../web3/UI/EthereumChainBoundary'
+import { ActionButtonPromise } from '../../../extension/options-page/DashboardComponents/ActionButton'
+
+export interface SavingsFormProps {
+    chainId: number
+    selectedProtocol: number
+    tab: TabType
+    onClose?: () => void
+    onSwapDialogOpen?: () => void
+}
+
+export function SavingsForm({ chainId, selectedProtocol, tab, onClose, onSwapDialogOpen }: SavingsFormProps) {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const protocol = SavingsProtocols[selectedProtocol]
+    const targetChainId = chainId
+
+    const web3 = useWeb3(false, targetChainId)
+    const account = useAccount()
+
+    const [inputAmount, setInputAmount] = useState('')
+
+    const { value: nativeTokenBalance, loading: loadingNativeTokenBalance } = useFungibleTokenBalance(
+        EthereumTokenType.Native,
+        '',
+        targetChainId,
+    )
+
+    //#region form controls
+    const formattedBalance = leftShift(Number.parseInt(nativeTokenBalance || '0', 16), 18)
+    const tokenAmount = rightShift(inputAmount || '0', 18)
+    //#endregion
+
+    //#region UI logic
+    // validate form return a message if an error exists
+    const validationMessage = useMemo(() => {
+        if (tokenAmount.isZero()) return t('plugin_trader_error_amount_absence')
+        if (isLessThan(inputAmount, 0)) return t('plugin_trade_error_input_amount_less_minimum_amount')
+
+        const tokenBalance =
+            (tab === TabType.Deposit
+                ? nativeTokenBalance
+                : rightShift(protocol.balance, 18).integerValue().toFixed()) || '0'
+
+        if (isLessThan(tokenBalance, tokenAmount)) {
+            return t('plugin_trader_error_insufficient_balance', {
+                symbol: tab === TabType.Deposit ? protocol.base : protocol.pair,
+            })
+        }
+
+        return ''
+    }, [inputAmount, tokenAmount.toFixed(), formattedBalance])
+    //#endregion
+
+    const needsSwap = protocol.name === 'Lido' && tab === 'withdraw'
+
+    return (
+        <div className={classes.containerWrap}>
+            <Typography variant="h3" className={classes.title}>
+                <img src={IconURLS.lido} className={classes.titleImage} />
+                {tab === TabType.Deposit ? 'Stake' : 'Withdraw'} {protocol.name}
```

Comment:
Please read the [i18n-guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and trun dead code textures to i18n fields.